### ISSUE TITLE
Having problems mapping two static enums

### ISSUE THREAD

**Issue by sabellefallEFC** [AutoMapper/AutoMapper#763]
I've been using AutoMapper successfully for several weeks, but I'm stuck with this one problem that is throwing a AutoMapper.AutoMapperMappingException.

I'm trying to map an Enum from a static Class : TypePayeurEnum.TypePayeur. The source and destination classes are using the exact same enum type and attribute names.

In both source and destination classes, I have this : 
```
    public TypePayeurEnum.TypePayeur typePayeur
        {
            get { return TYPE_PAYEUR; } 
            set { TYPE_PAYEUR = value;  }
        }
```

Here are excerpts of the Exception message I get : 

Missing type map configuration or unsupported mapping. 
Mapping types:
RuntimeType -> TypePayeur
System.RuntimeType -> Red.Modele.General.Enum.TypePayeurEnum+TypePayeur
{Trying to map RuntimeType to TypePayeur.}

So it seems to me that for some reason, AutoMapper is trying to map the RuntimeType to the TypePayeur enum. I don't get why ! 
I'm wondering if this problem has something to do with the fact that this enum comes from a static class ? is there something specific I must do in this case ? 

As for the CreateMap configuration it's very simple since I'm not using any option at all. All my attributes in the source and destination classes have the same names and types so I'm simply using the `CreateMap<source,dest>()`... and it works well for all the other classes (they don't have enums in them) Trying to solve this problem I've tried to use `ForMember` to force AutoMapper map the right thing but it didn't work.

Any help would be appreciated, I might be missing something really simple...

**Comment by lbargaoanu** [comment]
I don't see it. Please provide a failing test.

**Comment by sabellefallEFC** [comment]
I created a test to map two static enums and it passed. 
So there's a problem in my environment, I guess I'll keep investigating..

**Comment by lbargaoanu** [comment]
Can you please close this?

**Comment by sabellefallEFC** [comment]
After long hours of searching and trial and error, it works now.
It seems the problem is similar to this other issue : #291 : there are problems with properties that include "type" in their name. I renamed the property so that it doesn't include "type" anymore, but it didn't work until I renamed every single occurrence, including the param names in the constructor.

I saw in the #291 that there could be a mismatch with a GetType() method but there was not such method in my class. I had tried to rename the getters so that they don't include "type" at all but this alone wasn't enough.

I'm sure there is a rational explanation to all this and would be interested to know if someone finds it someday. At least now I can go on with my project.
